File: treebolicGlue/src/main/java/treebolic/glue/EventListener.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>
 */

package treebolic.glue;

/**
 * Event listener
 *
 * @author Bernard Bou
 */
public abstract class EventListener implements treebolic.glue.iface.EventListener
{
	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onDown(int x, int y, boolean rotate);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onUp(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onDragged(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onHover(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onSelect(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onFocus(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onLink(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onMenu(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public boolean onMount(int x, int y);

	@SuppressWarnings("UnusedReturnValue")
	@Override
	abstract public void onScale(float mapScale, float fontScale, float imageScale);
}

File: treebolicGlue/src/main/java/treebolic/glue/EventListener.kt
/*
 * Copyright (c) 2019-2023. Bernard Bou
 */
package treebolic.glue

import treebolic.glue.iface.EventListener

/**
 * Event listener
 *
 * @author Bernard Bou
 */
abstract class EventListener : EventListener {

    abstract override fun onDown(x: Int, y: Int, rotate: Boolean): Boolean

    abstract override fun onUp(x: Int, y: Int): Boolean

    abstract override fun onDragged(x: Int, y: Int): Boolean

    abstract override fun onHover(x: Int, y: Int): Boolean

    abstract override fun onSelect(x: Int, y: Int): Boolean

    abstract override fun onFocus(x: Int, y: Int): Boolean

    abstract override fun onLink(x: Int, y: Int): Boolean

    abstract override fun onMenu(x: Int, y: Int): Boolean

    abstract override fun onMount(x: Int, y: Int): Boolean

    abstract override fun onScale(mapScale: Float, fontScale: Float, imageScale: Float)
}
